EIPS/eip-4944.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 4944
+title: NFT with wallet
+description: An ERC721-compatible single-token NFT
+author: Victor Munoz (@victormunoz)
+discussions-to: https://ethereum-magicians.org/t/erc721-minting-only-one-token/8602/2
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-25
+requires: 721
+---
+
+## Abstract
+The following describes standard functions for an ERC721 compatible contract that can only only one minted token.
+This allows an NFT to be associated uniquely with a single contract address.
+
+## Motivation
+If the ERC721 was modified to mint only 1 token (per contract), then the contract address could be identified uniquely with that minted token (instead of the tuple contract address + token id, as ERC721 requires).
+This change would enable automatically all the capabilities of composable tokens ERC-998 (own other ERC721 or ERC20) natively without adding any extra code, just forbidding to mint more than one token per deployed contract.
+Then the NFT minted with this contract could operate with his "budget" (the ERC20 he owned) and also trade with the other NFTs he could own. Just like an autonomous agent, that could decide what to do with his properties (sell his NFTs, buy other NFTs, etc).
+
+The first use case that is devised is for value preservation. Digital assets, as NFTs, have value that has to be preserved in order to not be lost. If the asset has its own budget (in other ERC20 coins), could use it to autopreserve itself.
+
+## Specification
+The constructor should mint the unique token of the contract, and then the mint function should add a restriction to avoid further minting.
```

Comment:
What is the `tokenId` of the minted token? Does it matter?

Reply:
It is just there for compatibility with current ERC721 platforms.
The value should be 0 as suggested in the implementation section.